### PR TITLE
Login

### PR DESCRIPTION
**Pull request by FullOfPit**

### CODE REVIEW

File: backend/src/main/java/com/example/backend/appuser/AppUserController.java [hunk 1]
```diff
@@ -8,7 +8,7 @@
 import java.util.Optional;
 
 @RestController
-@RequestMapping("/api/appuser")
+@RequestMapping("/api/app-user")
```
Comment (yellowFloppyDrive) [inline backend/src/main/java/com/example/backend/appuser/AppUserController.java]: Resource REST endpoints should always be written in plural. `/api/app-users`

File: backend/pom.xml [hunk 2]
```diff
@@ -17,6 +17,7 @@
         <java.version>19</java.version>
         <sonar.organization>fullofpit</sonar.organization>
         <sonar.host.url>https://sonarcloud.io</sonar.host.url>
+        <sonar.exclusions>**/config/**</sonar.exclusions>
```
Comment (yellowFloppyDrive) [inline backend/pom.xml:20]: Not described in the PR changelist

File: backend/src/main/java/com/example/backend/risk/Risk.java [hunk 3]
```diff
@@ -19,9 +19,9 @@ public class Risk {
     private String riskDescription;
     private List<String> riskReductionMeasures;
 
-    private int healthHazard;
-    private int probability;
-    private int frequency;
+    private HealthHazard healthHazard;
+    private Probability probability;
```
Comment (yellowFloppyDrive) [inline backend/src/main/java/com/example/backend/risk/Risk.java]: Be careful when using too many enums. This leads to too many ifs later on. Probability sounds like a double value 0.0 to 1.0. I don't know the exact intention of this field. If this does not apply, please ignore this comment.

File: backend/src/main/java/com/example/backend/risk/example [hunk 4]
```diff
@@ -5,7 +5,7 @@
         "riskName": "Test Name",
```
Comment (yellowFloppyDrive) [inline backend/src/main/java/com/example/backend/risk/example]: Static and resource files should be placed under resources. This file also does not seem to be used anywhere. Please remove this!

File: frontend/src/App.tsx [hunk 5]
```diff
@@ -1,26 +1,37 @@
...
+      </div>
+
+
   );
 }
 
 export default App;
+
+{
+    /*
```
Comment (yellowFloppyDrive) [inline frontend/src/App.tsx]: We do not want commented out code to be merged to main!

File: frontend/src/components/Authentication.tsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+    const navigate = (
+        <Navigate
+            to={isReady && !user ?
+                `/login?redirect=${encodeURIComponent(location.pathname + location.search)}`
+                : "/"}/>
+    );
+
+    if (isReady && user) {
+        return <>{children}</>
```
Comment (yellowFloppyDrive) [inline frontend/src/components/Authentication.tsx]: Missing semi-colons at the end of the line!

File: frontend/src/components/Logout.tsx [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    const logout = useCallback(async () => {
+        await axios.get("/api/app-user/logout");
+        navigate("/login?redirect=" + encodeURIComponent(location.pathname + location.search));
+        window.document.cookie = "";
+        window.localStorage.clear();
+    }, [location, navigate]);
+
+    return (
+        <button onClick={logout}>Logout</button>
+    )
```
Comment (yellowFloppyDrive) [inline frontend/src/components/Logout.tsx]: Missing semi-colon

File: frontend/src/App.tsx [hunk 8]
```diff
@@ -1,26 +1,37 @@
 import React from 'react';
-import logo from './logo.svg';
 import './App.css';
+import {BrowserRouter} from "react-router-dom";
+import Root from "./pages/Root";
+
 
 function App() {
+
```
Comment (yellowFloppyDrive) [inline frontend/src/App.tsx:7]: Unnecessary empty lines

File: frontend/src/App.tsx [hunk 9]
```diff
@@ -1,26 +1,37 @@
...
+          </BrowserRouter>
+      </div>
+
+
   );
 }
 
 export default App;
+
+{
```
Comment (yellowFloppyDrive) [inline frontend/src/App.tsx]: What are these brackets used for?

File: frontend/src/components/NoAuth.tsx [hunk 10]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import useAuth from "../hook/useAuth";
+import {Navigate} from "react-router-dom";
+
+export default function NoAuth({
+    children,
+    redirect
+                               }:{
```
Comment (yellowFloppyDrive) [inline frontend/src/components/NoAuth.tsx]: What are these space characters used for?

File: frontend/src/hook/useAuth.tsx [hunk 11]
```diff
@@ -0,0 +1,23 @@
+import {useEffect, useState} from "react";
+import axios from "axios";
+
+
```
Comment (yellowFloppyDrive) [inline frontend/src/hook/useAuth.tsx]: Unnecessary empty line

File: frontend/src/index.css [hunk 12]
```diff
@@ -11,3 +11,10 @@ code {
   font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
     monospace;
 }
+
+.ButtonGeneral {
+  height: 1.7rem;
+  width: 10.0rem;
+
```
Comment (yellowFloppyDrive) [inline frontend/src/index.css]: Unnecessary empty line

File: frontend/src/pages/Dashboard.tsx [hunk 13]
```diff
@@ -1,5 +1,16 @@
+import Logout from "../components/Logout";
+
+
```
Comment (yellowFloppyDrive) [inline frontend/src/pages/Dashboard.tsx]: Unnecessary empty line

**Review by yellowFloppyDrive — CHANGES_REQUESTED** [review]
DO NOT NAME YOUR BRANCHES WITH CAPITAL LETTERS!!!

File: backend/src/main/java/com/example/backend/appuser/AppUserController.java (hunk 1 above)
Comment (FullOfPit) [reply backend/src/main/java/com/example/backend/appuser/AppUserController.java]: _Changed_

File: backend/pom.xml (hunk 2 above)
Comment (FullOfPit) [reply backend/pom.xml:20]: Oh damn, sorry! Won't happen again

File: backend/src/main/java/com/example/backend/risk/Risk.java (hunk 3 above)
Comment (FullOfPit) [reply backend/src/main/java/com/example/backend/risk/Risk.java]: Hmmm, well, all 3 of these are discrete ints between and including 1 and 4, though they often described with their respective names... So I'm not entirely sure whether I should pick ints or enums. Your suggestion would be using ints, it take it?

File: backend/src/main/java/com/example/backend/risk/example (hunk 4 above)
Comment (FullOfPit) [reply backend/src/main/java/com/example/backend/risk/example]: Fair enough
_Deleted_

File: frontend/src/App.tsx (hunk 5 above)
Comment (FullOfPit) [reply frontend/src/App.tsx]: _Deleted_

File: frontend/src/components/Authentication.tsx (hunk 6 above)
Comment (FullOfPit) [reply frontend/src/components/Authentication.tsx]: _Changed_

File: frontend/src/components/Logout.tsx (hunk 7 above)
Comment (FullOfPit) [reply frontend/src/components/Logout.tsx]: _Changed_

File: frontend/src/App.tsx (hunk 8 above)
Comment (FullOfPit) [reply frontend/src/App.tsx:7]: _Changed_

File: frontend/src/App.tsx (hunk 9 above)
Comment (FullOfPit) [reply frontend/src/App.tsx]: Remnants of that commented out code, I guess
_Changed_

File: frontend/src/components/NoAuth.tsx (hunk 10 above)
Comment (FullOfPit) [reply frontend/src/components/NoAuth.tsx]: Helps me get the syntax right, but I forgot to take them out here!
_Changed_

File: frontend/src/hook/useAuth.tsx (hunk 11 above)
Comment (FullOfPit) [reply frontend/src/hook/useAuth.tsx]: _Changed_

File: frontend/src/index.css (hunk 12 above)
Comment (FullOfPit) [reply frontend/src/index.css]: _Changed_

File: frontend/src/pages/Dashboard.tsx (hunk 13 above)
Comment (FullOfPit) [reply frontend/src/pages/Dashboard.tsx]: _Changed_

File: backend/src/main/java/com/example/backend/risk/Risk.java (hunk 3 above)
Comment (FullOfPit) [reply backend/src/main/java/com/example/backend/risk/Risk.java]: _Changed_

File: backend/src/main/java/com/example/backend/appuser/AppUserController.java (hunk 1 above)
Comment (yellowFloppyDrive) [reply backend/src/main/java/com/example/backend/appuser/AppUserController.java]: [https://restfulapi.net/resource-naming/](https://restfulapi.net/resource-naming/)

File: backend/pom.xml (hunk 2 above)
Comment (yellowFloppyDrive) [reply backend/pom.xml:20]: Well need to add this to the changelist above. You can still edit your PR

**Review by yellowFloppyDrive — APPROVED** [review]
lgtm

**Pull request merged**